src/main/java/org/cryptomator/cryptofs/CryptoFileSystemModule.java

```diff
@@ -37,6 +39,20 @@ public Cryptor provideCryptor(CryptorProvider cryptorProvider, @PathToVault Path
 		}
 	}
 
+	private String generateFileIdIfPossible(byte[] fileBytes) {
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-256");
+			byte[] truncatedDigest = md.digest(fileBytes);
+			StringBuilder sb = new StringBuilder();
+			for (int i = 0; i < 4; i++) { // we only need the first 4 bytes, since each is translated into two characters
+				sb.append(String.format("%02x", truncatedDigest[i]));
+			}
+			return sb.toString().substring(0, 7);
+		} catch (NoSuchAlgorithmException e) {
+			return "";
```

Comment:
```suggestion
			throw new IllegalStateException(e);
```

Quoting [MessageDigest JavaDoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/MessageDigest.html):

> Every implementation of the Java platform is required to support the following standard MessageDigest algorithms:
>
> MD5
> SHA-1
> SHA-256

We might even be able to throw an AssertionError, but some static code analysis tools might not like this.